### PR TITLE
Refactor tracer test to use rhel_contenthost fixture

### PR DESCRIPTION
**Pull request by rplevka**
I refactored the test for tracer integration.
It is now using `rhel_contenthost` fixture so it is parametrized for all supported rhel versions.

Also, the logic of the test was fixed as the original version wasn't testing the host-side service state properly.

Thanks @ogajduse for writing the custom, mock service for the test.

tracer is supported since rhel7, so we're excluding rhel6 scenario:
```
$ pytest -k tracer_list  tests/foreman/api/test_host.py --collect-only
==== test session starts ====
platform linux -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/rplevka/work/rplevka/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, reportportal-5.0.8, mock-3.6.1, xdist-2.5.0, ibutsu-2.0.2
collected 65 items / 62 deselected / 3 selected                                                                       

<Package api>
  <Module test_host.py>
    <Class TestHostTraces>
        <Function test_positive_tracer_list_and_resolve[rhel8]>
        <Function test_positive_tracer_list_and_resolve[rhel9]>
        <Function test_positive_tracer_list_and_resolve[rhel7]>

==== 3/65 tests collected (62 deselected) in 0.19s ====

```

results (rhel9 scenario failure is expected, since we don't have the rhel9 client repo yet):

```
$ pytest -k tracer_list  tests/foreman/api/test_host.py -n3
==== test session starts ====
platform linux -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/rplevka/work/rplevka/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, reportportal-5.0.8, mock-3.6.1, xdist-2.5.0, ibutsu-2.0.2
gw0 [3] / gw1 [3] / gw2 [3]
..E                                                                                                             [100%]
==== ERRORS ====
...
'errors': ["404, message='Not Found', url=URL('https://dogfood...')
...

==== 2 passed, 1 error in 647.86s (0:10:47) ====
```

### CODE REVIEW

File: tests/foreman/api/test_host.py [hunk 1]
```diff
@@ -32,11 +32,65 @@
...
+from robottelo.config import settings
 
 from robottelo import datafactory
 from robottelo.api.utils import promote
 from robottelo.config import get_credentials
 
+@pytest.fixture(scope='function')
+def tracer_host(default_sat, module_org, rhel_contenthost):
+    rhelver = rhel_contenthost.os_version.major
+    if settings.repos['SATELLITE_VERSION_UNDR'].startswith('7'):
```
Comment (mshriver) [inline tests/foreman/api/test_host.py:44]: I think you should use `default_sat.os_version.major` here - also `master` branch right now is for 7.0, why is this check necessary?

File: tests/foreman/api/test_host.py [hunk 2]
```diff
@@ -32,11 +32,65 @@
...
+    prod = entities.Product(organization=module_org, name=f'rhel{rhelver}_{gen_string("alpha")}').create()
+    sattools_repo = entities.Repository(
+        organization=module_org,
+        product=prod,
+        content_type='yum',
+        url=settings.repos[repo_name][f'RHEL{rhelver}']
+    ).create()
+    sattools_repo.sync()
+    subs = entities.Subscription(organization=module_org, name=prod.name).search()
+    assert len(subs) > 0, f'Subscription for sat client product: {prod.name} was not found.'
```
Comment (mshriver) [inline tests/foreman/api/test_host.py:57]: `>0` is redundant here, non-zero integers are truthy.

File: tests/foreman/api/test_host.py (hunk 1 above)
Comment (ogajduse) [reply tests/foreman/api/test_host.py:44]: True, this logic is not needed here if we are not going to backport this patch to `.z` branches.

File: tests/foreman/api/test_host.py [hunk 3]
```diff
@@ -32,11 +32,65 @@
...
 from robottelo.config import get_credentials
 
+@pytest.fixture(scope='function')
+def tracer_host(default_sat, module_org, rhel_contenthost):
+    rhelver = rhel_contenthost.os_version.major
+    if settings.repos['SATELLITE_VERSION_UNDR'].startswith('7'):
+        repo_name = 'SATCLIENT_REPO'
+    else:
+        repo_name = 'SATTOOLS_REPO'
+    prod = entities.Product(organization=module_org, name=f'rhel{rhelver}_{gen_string("alpha")}').create()
```
[the remark this answers is not in the document]
Comment (rplevka) [reply tests/foreman/api/test_host.py:48]: The module-scoped repo fixture based on supportability is definitely something that comes to mind, but it does not honor the potential version filter test marks, so they potentially crate and sync much more than the module really requires.

Some research should be done around a possibility to create a test collection hook function that would somehow check for these marks on all collected test for a module (that also use the repo fixture), combine them and finally create and sync the appropriate repositories.

Till then, I think we're fine with going with this. The tools/client repos are small and the sync time does not add too much overhead (Other tests do much worse).

File: tests/foreman/api/test_host.py (hunk 1 above)
Comment (rplevka) [reply tests/foreman/api/test_host.py:44]: Actually, I think we can backport this fixture and the relevant tests to `6.10.z`, but we will use the appropriate repo name there (`SATTOOLS_REPO`).
Good catch.

**Review by tstrych — APPROVED** [review]
ACK
Please address the above comments appropriately, 
Thank you for adding mocking service repo, 
Thanks for line comments that differentiate parts of the code.
This is introducing new fixture, but we have only 2 tests with tracer, It would be nice to convert also the other one. If this is not your priority I understand, but please create an issue and assign it to me so I will not forget. No need for code duplicity.

**Comment by rplevka** [comment]
I addressed the comments including the 1 that came out of our discussion with @ogajduse -
I broken down the big, local fixture to the global ones.

```
=== 2 passed, 1 error in 612.86s (0:10:12) ===
```

**Comment by rplevka** [comment]
> This is introducing new fixture, but we have only 2 tests with tracer, It would be nice to convert also the other one.

@ogajduse asked me about the same, so now, with such a great demand, I'm feeling obliged to convert the other test as well ;)
I'm done with it and the tests are running ATM, one more force-push coming up soon

**Comment by rplevka** [comment]
Very well,
both API and CLI tests are now converted:

```
$ pytest -k test_positive_tracer_list_and_resolve tests/foreman/cli/test_host.py -n3
==== test session starts ====
platform linux -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/rplevka/work/rplevka/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, reportportal-5.0.8, mock-3.6.1, xdist-2.5.0, ibutsu-2.0.2
gw0 [3] / gw1 [3] / gw2 [3]
..E                                      
===== 2 passed, 3 warnings, 1 error in 609.70s (0:10:09) =====

**Review by ogajduse — APPROVED** [review]
@rplevka Thanks a ton for this PR! This is a nice example for all content host component owners who can use this PR as an example. I have no objections as we discussed it on a call.
I also left some TODOs for @tstrych

**Review by tstrych — APPROVED** [review]
This is more that you sign for. Thanks, I highly appreciate the new mock service, `wait_for ` handlers,  bulk subscription missing part in test logic, and all of the testing with 4 RHEL versions.

Tests are passing, and this PR should be on my plate. Please bear that in mind before requesting additional changes. If comment is not required but nice to have pleace leave the comment with TODO, and I will address those problems in other PR.

**Review by pondrejk — APPROVED** [review]

File: tests/foreman/cli/test_host.py [hunk 4]
```diff
@@ -2598,29 +2655,43 @@ def test_positive_dump_enc_yaml(default_sat):
...
+    that tracer shows and resolves the problem. The test works with a package specified
+    in settings. This package is expected to install a systemd service which is expected
+    to log its version to /var/log/{package}/service.log.
+    The rpm is not supposed to start the service upon install.
 
     :id: 81c83a2c-4b9d-11ec-a5b3-98fa9b6ecd5a
 
-    :expectedresults: Tracer resolved the problem, the downgraded service was restarted
+    :expectedresults: Tracer detected and resolved the problem, the affected service
+    was restarted
```
[the remark this answers is not in the document]
Comment (tstrych) [reply tests/foreman/cli/test_host.py:2669]: I tested it for you, this works.
```suggestion
        was restarted
```

**Pull request merged**